clouddriver-web/src/main/groovy/com/netflix/spinnaker/clouddriver/configuration/CustomConfigClientConfiguration.groovy

```diff
@@ -0,0 +1,107 @@
+import org.springframework.cloud.config.client.ConfigServicePropertySourceLocator
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import org.springframework.core.env.Environment
+import org.springframework.http.client.ClientHttpRequestFactory
+import org.springframework.http.client.HttpComponentsClientHttpRequestFactory
+import org.springframework.web.client.RestTemplate
+
+import javax.net.ssl.KeyManagerFactory
+import javax.net.ssl.SSLContext
+import java.security.KeyStore
+import java.security.SecureRandom
+
+/**
+ * Configures Spring Cloud Config Client to speak to a Config Server over an SSL encrypted channel.
+ *
+ * NOTE: By default, spring.cloud.config.enabled=true activates Config Client, but WITHOUT SSL. TO
+ *       have Config Client with SSL, use config-client.ssl.enabled=true instead.
+ *
+ * For property settings, see commented out settings in bootstrap.yml.
+ *
+ * NOTE: This is independent of serving requests via SSL.
+ */
+@Configuration
+@ConditionalOnProperty('config-client.ssl.enabled')
+@EnableConfigurationProperties(ConfigClientSslProperties)
+class CustomConfigClientConfiguration {
+
+	@Autowired
+	ConfigClientSslProperties configClientSslProperties
+
+
+	@Bean
+	ConfigClientProperties configClientProperties(Environment env) {
+		ConfigClientProperties configClientProperties = new ConfigClientProperties(env)
+		configClientProperties.enabled = false
+		configClientProperties
+	}
+
+	@Bean
+	ConfigServicePropertySourceLocator configServicePropertySourceLocator(ConfigClientProperties clientProperties,
+																		  RestTemplate restTemplate) {
+		ConfigServicePropertySourceLocator configServicePropertySourceLocator = new ConfigServicePropertySourceLocator(clientProperties)
+		configServicePropertySourceLocator.restTemplate = restTemplate
+		configServicePropertySourceLocator
+	}
+
+	@Bean
+	RestTemplate restTemplate(ClientHttpRequestFactory clientHttpRequestFactory) {
+		new RestTemplate(clientHttpRequestFactory)
+	}
+
+	@Bean
+	ClientHttpRequestFactory clientHttpRequestFactory(HttpClient httpClient) {
+		new HttpComponentsClientHttpRequestFactory(httpClient)
+	}
+
+	@Bean
+	HttpClient httpClient() {
+		KeyManagerFactory keyManagerFactory = KeyManagerFactory.getInstance("SunX509")
+		KeyStore trustStore = KeyStore.getInstance(configClientSslProperties.ssl.keyStoreType)
+
+		def keyStoreFile = new File(configClientSslProperties.ssl.keyStore)
+
+		if (keyStoreFile.exists()) {
+			trustStore.load(keyStoreFile.newInputStream(), configClientSslProperties.ssl.keyStorePassword.toCharArray())
+			keyManagerFactory.init(trustStore, configClientSslProperties.ssl.keyStorePassword.toCharArray())
+		} else {
+			throw new RuntimeException("Cannot find resource: ${keyStoreFile.absolutePath}")
+		}
+
+		SSLContext sslContext = SSLContexts.custom().loadTrustMaterial(trustStore, new TrustSelfSignedStrategy()).build()
```

Comment:
have a look at https://github.com/spinnaker/kork/blob/master/kork-web/src/main/groovy/com/netflix/spinnaker/config/OkHttpClientConfiguration.groovy#L75 for how the SSL context init happens

that properly configures both the keystore and truststore so there is explicit trust of both the client and server